internal/daemon/daemon.go

```diff
@@ -493,23 +493,7 @@ func (d *Daemon) initServer(resources ...rest.Resources) *http.Server {
 
 // StartAPI starts up the admin and consumer APIs, and generates a cluster cert
 // if we are bootstrapping the first node.
-func (d *Daemon) StartAPI(ctx context.Context, bootstrap bool, initConfig map[string]string, newConfig *trust.Location, joinAddresses ...string) error {
-	if newConfig != nil {
```

Comment:
There was the scenario where if the cluster isn't yet initialized (pre-init) and you query the cluster status enpoint you will get an error as the node address couldn't be parsed as it wasn't yet set.

Would this return this bug?

Reply:
It shouldn't, the behaviour of this `if` block is still preserved. 
We call `StartAPI` in 3 places:
* restart - (uses the `else` case), it was actually already redundant since we call `config.Load` [here](https://github.com/canonical/microcluster/blob/a4c5273899ac294750d63089f81f78fd1894df67/internal/daemon/daemon.go#L418-L426)
* join - it's still called before we run `StartAPI` [here](https://github.com/canonical/microcluster/blob/6dc1ebc092f502d6e71a22dad9ff2ef5977323aa/internal/rest/resources/control.go#L63)
* bootstrap - it's still called before we run `StartAPI` [here](https://github.com/canonical/microcluster/blob/6dc1ebc092f502d6e71a22dad9ff2ef5977323aa/internal/rest/resources/control.go#L63)

We also support returning an empty addess from the status endpoint, so it shouldn't be an issue anyway.

Reply:
Thanks for pointing me to https://github.com/canonical/microcluster/blob/a4c5273899ac294750d63089f81f78fd1894df67/internal/rest/resources/api_1.0.go#L21-L25 which was what I meant originally :)